ocaml/lambda/simplif.ml

```diff
@@ -901,7 +912,10 @@ let simplify_local_functions lam =
         Hashtbl.remove slots id
     | Lfunction lf as lam ->
         check_static lf;
-        Lambda.shallow_iter ~tail ~non_tail lam
+        let old_function_scope = !current_function_scope in
+        current_function_scope := lf.body;
+        Lambda.shallow_iter ~tail ~non_tail lam;
```

Comment:
Since `lam = Lfunction lf`, then `shallow_iter ~tail ~non_tail lam` is the same as `non_tail lf.body`. This is then exactly the same code sequence as above in Llet, so could be pulled out to a separate function below (like `region`)

Reply:
Thanks, I've pushed a patch doing that